### DISCUSSION TITLE
Bob `v2.4.31` is out - Reference variables in compile parameters, use wildcarding to override rules for certain objects types, and more fixes!

### DISCUSSION THREAD

**Discussion by SanjulaGanepola** [IBM/ibmi-tobi#355] [Announcements]
Bob `v2.4.31` is out with major enhancements and bug fixes!

### Enhancements
- Support referencing variables in compiler parameter overrides (https://ibm.github.io/ibmi-bob/#/prepare-the-project/rules.mk?id=variables)
  ```make
  PROJECT_TGTRLS := *PRV
  
  PRO200.MODULE: private TGTRLS := $(PROJECT_TGTRLS)
  PRO200.MODULE: PRO200.RPGLE
  
  VAT.MODULE: private TGTRLS := $(PROJECT_TGTRLS)
  VAT.MODULE: VAT.RPGLE
  ```
- Support wildcarding to override the rule to make certain types of objects (https://ibm.github.io/ibmi-bob/#/prepare-the-project/rules.mk?id=wildcarding)
  ```make
  FILE_TEXT := SAMPLE1
  MOD_TEXT := SAMPLE2
  
  %.FILE: private TEXT := $(FILE_TEXT)
  %.MODULE: private TEXT := $(MOD_TEXT)
  
  %.FILE: %.TABLE DEP1.FILE DEP2.FILE
  %.MODULE: %.RPGLE
  ```
### Fixes
- Update `RUNSQLFLAGS` to recursive assignment to allow for overridden `TGTRLS`
- Fix logs not creating on initial deploy
- Support index file extension
- Fix typos in error messages
- Fix `CHGOBJD` for text description to use target object name instead of source name

Full Changelog: [2.4.30..2.4.31](https://github.com/IBM/ibmi-bob/compare/v2.4.30...v2.4.31)

**Comment by NickLitten** [comment]
This is terrific feature - thankyou!

Is there a way to set a default, at a project level and have it default without having to define the over-ride for all compiles?

For example, I want to set the default TGTRLS to *PRV for every compile in my build.

**Reply by SanjulaGanepola** [reply]
@edmundreinhardt Are you aware of any way to do this right now or could this be a potential enhacement?

**Comment by edmundreinhardt** [comment]
Actually, there is a new feature that let's you set default for a member type just added a month ago:

https://ibm.github.io/ibmi-bob/#/prepare-the-project/rules.mk?id=overriding-compile-settings
